sdk/core/core-util/src/delay.ts

```diff
@@ -28,38 +27,24 @@ export interface DelayOptions {
  */
 export function delay(timeInMs: number, options?: DelayOptions): Promise<void> {
   return new Promise((resolve, reject) => {
-    let timer: ReturnType<typeof setTimeout> | undefined = undefined;
-    let onAborted: (() => void) | undefined = undefined;
-
     const rejectOnAbort = (): void => {
-      return reject(new AbortError(options?.abortErrorMsg ?? StandardAbortMessage));
+      reject(new AbortError(options?.abortErrorMsg ?? StandardAbortMessage));
     };
-
     const removeListeners = (): void => {
-      if (options?.abortSignal && onAborted) {
-        options.abortSignal.removeEventListener("abort", onAborted);
-      }
+      options?.abortSignal?.removeEventListener("abort", onAbort);
     };
-
-    onAborted = (): void => {
-      if (isDefined(timer)) {
-        clearTimeout(timer);
-      }
```

Comment:
[Passing `undefined` to `clearTimeout` does nothing](https://developer.mozilla.org/en-US/docs/Web/API/clearTimeout#notes) and it is not gonna happen anyway.